modules/product/src/main/java/com/opengamma/strata/product/credit/CdsTradeForCalibration.java

```diff
@@ -0,0 +1,350 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.credit;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanBuilder;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.product.Trade;
+import com.opengamma.strata.product.TradeInfo;
+
+/**
+ * A trade in a single-name credit default swap (CDS) used for credit curve calibration. 
+ * <p>
+ * The CDS trade and market quote are stored in this class.
+ * {@link CdsTrade} and {@code ResolvedCdsTrade} should be used for pricing.
+ */
+@BeanDefinition(builderScope = "private")
+public final class CdsTradeForCalibration
+    implements Trade, ImmutableBean, Serializable {
+
+  /**
+   * The CDS trade.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final CdsTrade trade;
```

Comment:
To avoid code like `getTrade().getTrade()` this should be named `underlyingTrade`.